gtest/src/mailbox.rs

```diff
@@ -0,0 +1,99 @@
+use crate::{manager::ExtManager, CoreLog, Log, RunResult};
+use codec::Encode;
+use gear_core::message::{Dispatch, DispatchKind, StoredMessage};
+use gear_core::{
+    ids::{MessageId, ProgramId},
+    message::Payload,
+};
+use std::cell::RefCell;
+
+pub struct Mailbox<'a> {
+    manager_reference: &'a RefCell<ExtManager>,
+    program_id: ProgramId,
+}
+
+impl<'a> Mailbox<'a> {
+    pub(crate) fn new(
+        program_id: ProgramId,
+        manager_reference: &'a RefCell<ExtManager>,
+    ) -> Mailbox<'a> {
+        Mailbox {
+            program_id,
+            manager_reference,
+        }
+    }
+
+    pub fn contains(&self, log: Log) -> bool {
+        self.manager_reference
+            .borrow_mut()
+            .actor_to_mailbox
+            .get(&self.program_id)
+            .expect("No mailbox with such program id")
```

Comment:
This expect should be infallible due to existence of the mailbox structure. So need to replace the expect message or add here get_or_insert_with_default
```suggestion
            .get(&self.program_id)
            .expect("No mailbox with such program id")
```